bsp/nrf/lh2.c

```diff
@@ -263,102 +789,141 @@ void db_lh2_init(db_lh2_t *lh2, const gpio_t *gpio_d, const gpio_t *gpio_e) {
+    lh2->raw_data[sweep][temp_selected_polynomial >> 1].selected_polynomial = temp_selected_polynomial;
+    lh2->raw_data[sweep][temp_selected_polynomial >> 1].bits_sweep          = temp_bits_sweep;
+    lh2->timestamps[sweep][temp_selected_polynomial >> 1]                   = temp_timestamp;
+    lh2->data_ready[sweep][temp_selected_polynomial >> 1]                   = DB_LH2_RAW_DATA_AVAILABLE;
 }
 
 void db_lh2_process_location(db_lh2_t *lh2) {
-    if (lh2->state != DB_LH2_RAW_DATA_READY) {
-        return;
-    }
+
+    uint32_t lfsr_loc_temp;
 
     // compute LFSR locations and detect invalid packets
-    for (uint8_t location = 0; location < LH2_LOCATIONS_COUNT; location++) {
-        lh2->locations[location].selected_polynomial = lh2->raw_data[location].selected_polynomial;
-        // find location of the first data set by counting the LFSR backwards
-        lh2->locations[location].lfsr_location = _reverse_count_p(
-                                                     lh2->raw_data[location].selected_polynomial,
-                                                     lh2->raw_data[location].bits_sweep >> (47 - lh2->raw_data[location].bit_offset)) -
-                                                 lh2->raw_data[location].bit_offset;
-    }
+    for (uint8_t basestation = 0; basestation < LH2_BASESTATION_COUNT; basestation++) {
+        for (uint8_t sweep = 0; sweep < 2; sweep++) {
 
-    uint32_t tmp_locations[LH2_LOCATIONS_COUNT] = { 0 };
-    for (uint8_t location = 0; location < LH2_LOCATIONS_COUNT; location++) {
-        tmp_locations[location] = lh2->locations[location].lfsr_location;
-    }
+            // Check if this particular position has data point ready for sending
+            if (lh2->data_ready[sweep][basestation] == DB_LH2_RAW_DATA_AVAILABLE) {
 
-    // results from 2 LH2 were discovered - sort the two pairs
-    if (lh2->locations[0].lfsr_location > lh2->locations[1].lfsr_location) {
-        lh2->locations[0].selected_polynomial = lh2->raw_data[1].selected_polynomial;
-        lh2->locations[0].lfsr_location       = tmp_locations[1];
-        lh2->locations[1].selected_polynomial = lh2->raw_data[0].selected_polynomial;
-        lh2->locations[1].lfsr_location       = tmp_locations[0];
-    }
+                // Sanity check, make sure you don't start the LFSR search with a bit-sequence full of zeros.
+                if ((lh2->raw_data[sweep][basestation].bits_sweep >> (47 - lh2->raw_data[sweep][basestation].bit_offset)) == 0x000000) {
+                    // Mark the data as wrong and keep going
+                    lh2->data_ready[sweep][basestation] = DB_LH2_NO_NEW_DATA;
+                    continue;
+                }
+
+                // Copy the selected polynomial
+                lh2->locations[sweep][basestation].selected_polynomial = lh2->raw_data[sweep][basestation].selected_polynomial;
+                // Copmute and save the lsfr location.
+                lfsr_loc_temp = _reverse_count_p(
```

Comment:
`lfsr_loc_temp` is only used inside the scope of the double for loops, so no need to declare it outside.

```suggestion
                uint32_t lfsr_loc_temp = _reverse_count_p(
```

Reply:
added!

Question,
Isn't it slower to constantly re-declare a variable inside the double for loop, instead of declaring it once outside?

Reply:
thanks for the offline answer!
(not it's not slower, or maybe it's like 1 single assembly instruction slower)